main.tf

```diff
@@ -0,0 +1,112 @@
+terraform {
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = ">= 3.8"
+    }
+  }
+  backend "azurerm" {
+    resource_group_name  = "Softwire21_ThomasJohnston_ProjectExercise"
+    storage_account_name = "tfstate10453"
+    container_name       = "tfstate"
+    key                  = "terraform.tfstate"
+  }
+}
+provider "azurerm" {
+  features {}
+}
+variable "oauth_app_id" {
+  type      = string
+  sensitive = true
+}

```

Comment:
Not vital, but a standard practice is to pull these out into a `variables.tf` file (and similarly create an `outputs.tf` file)